Add documentation of how to turn off the Premium promotion button displayed in self-hosted Community instances of TinyMCE (#2502)
tinymce/tinymce-docs: modules/ROOT/pages/editor-premium-upgrade-promotion.adoc

= Premium upgrade promotion
:navtitle: {productname} Premium upgrade promotion
:description: Editor options related to turning the Premium promotion display off
:keywords:

== Premium upgrade promotion defaults

> When {productname} 6.2 or later is running in self-hosted Community mode, an *Upgrade* promotion button appears in the top-right of the editor menu bar by default. The button does note appear when {productname} is running in Community mode on the Tiny cloud.

Reviewer: ```suggestion
As part of the 6.2 release, self-hosted {productname} now features a promotional button labelled *Upgrade* which redirects to a link:{companyurl}/tinymce-self-hosted-premium-features/?utm_source=TinyMCE&utm_medium=SPAP&utm_campaign=SPAP&utm_id=editorreferral[promotion page], informing self-hosted users of the benefits of switching to {productname} cloud. This button is only visible in the self-hosted {communityname} version of the editor, and will not appear in any {cloudname} or self-hosted Premium versions of the editor.

The button can be removed by setting the `+promotion+` option to `+false+`.
```

Author: To quote my commit message:

> Added extra explanatory material regarding what the promotion button does.
> 
> Core text provided by @MitchCTiny. The edit and re-arrangement is my fault.

Reviewer 2: Should this really be linking with the exact same URL? That seems like a very bad idea will mess with Marketing analytics.

Author: I just copied-and-pasted, but I’m entirely happy to reduce that to the bare URL (ie, delete everything from the `?` to the end of the string).

Reviewer: Ah right sorry I just grabbed the link since I figured it would be good to acknowledge it on this page.

Author: And done: the URL is now bare.

Reviewer 3: Marketing may want a different set of query parameters for the docs, but I guess they'll have the docs in the referrer header which might be enough.

= Premium upgrade promotion
:navtitle: {productname} Premium upgrade promotion
:description: Editor options related to turning the Premium promotion display off

> :keywords:

Reviewer: Should there be any keywords here? E.g "Upgrade" "button" "promotion" "6.2". I don't even know if these keywords are used anywhere outside the repo but could be useful anyway in case someone needs to find this page in the repo.

Author: Good idea. Done.

Reviewer 2: @MitchCTiny just to let you know, these are actually used for SEO to some extent. They are rendered in the `<head>` like so:
```
<meta name="keywords" content="upgrade, promotion, premium, button">
```

== Premium upgrade promotion defaults

> When {productname} 6.2 or later is running in self-hosted Community mode, an *Upgrade* promotion button appears in the top-right of the editor menu bar by default. The button does note appear when {productname} is running in Community mode on the Tiny cloud.

Author: Typo corrected.

> When {productname} 6.2 or later is running in self-hosted Community mode, an *Upgrade* promotion button appears in the top-right of the editor menu bar by default. The button does note appear when {productname} is running in Community mode on the Tiny cloud.

Author: s/Tiny cloud/{cloudname}/.

Thanks for the extra eyes: I missed that one.

> When {productname} 6.2 or later is running in self-hosted Community mode, an *Upgrade* promotion button appears in the top-right of the editor menu bar by default. The button does note appear when {productname} is running in Community mode on the Tiny cloud.

Reviewer 2: Admittedly I also haven't seen us refer to them as "modes" before. I wonder if we'd be better saying something like "is running using a self-hosted Community distribution"? As they really aren't modes but different forms of distributions.

> When {productname} 6.2 or later is running in self-hosted Community mode, an *Upgrade* promotion button appears in the top-right of the editor menu bar by default. The button does note appear when {productname} is running in Community mode on the Tiny cloud.

Reviewer 2: Just to add other options, "entitlements" or "plans" could be used as well maybe?

> When {productname} 6.2 or later is running in self-hosted Community mode, an *Upgrade* promotion button appears in the top-right of the editor menu bar by default. The button does note appear when {productname} is running in Community mode on the Tiny cloud.

Author: Mode is gone. Distribution and Plan are in. (And I’ve re-written around these replacements.)

> This promotion button redirects to a link:{companyurl}/tinymce-self-hosted-premium-features/[promotion page], arguing the benefits of Community mode users switching from self-hosted to the {cloudname}.

Reviewer: ```suggestion
This promotion button redirects to a link:{companyurl}/tinymce-self-hosted-premium-features/[promotion page], showcasing the benefits of our wide variety of xref:plugins#premium-plugins[Premium Plugins] available for {productname}.
```
I wrote the wrong thing without reading the page, looks like the page mostly talks about premium plugins, not the cloud. Since the promotion only appears to self-hosted users on the core (free) plan, many of whom are from the open-source community, there is benefit in showcasing our premium features, without necessarily encouraging the switch to cloud.

> This promotion button redirects to a link:{companyurl}/tinymce-self-hosted-premium-features/?utm_source=TinyMCE&utm_medium=SPAP&utm_campaign=SPAP&utm_id=editorreferral[promotion page], arguing the benefits of Community mode users switching from self-hosted to the {productname} cloud.

Reviewer 2: ```suggestion
This promotion button redirects to a link:{companyurl}/tinymce-self-hosted-premium-features/?utm_source=TinyMCE&utm_medium=SPAP&utm_campaign=SPAP&utm_id=editorreferral[promotion page], arguing the benefits of Community mode users switching from self-hosted to the {cloudname}.
```

> When {productname} 6.2 or later is running in Premium mode, the *Upgrade* promotion button is disabled. And it is disabled when running on a Premium plan whether {productname} is running in the {cloudname} or as a self-hosted instance.

Author: > So I’ll just say it's a bit confusing, but trust your judgement.

FWIW, all of this is out of the journalist’s tradition of _tell ’em; tell ’em you told ’em; tell ’em again_. (And the triplet rhythm of that phrase is part of the structure.)

We _tell ’em_ in the 3rd paragraph on the page (before the code sample).

The first sentence in the admonition functions as the _tell ’em you told ’em_. And the construction you found confusing is the _tell ’em again_.

But it shouldn’t be confusing.

So I’ve re-cast it and gone all-in on straight up repetition (and leaned into the rhythm that comes from said straight up repetition). The new admonition is

**IMPORTANT: It is not necessary to set `promotion` to `false` when {productname} is running in Premium mode. Running {productname} in Premium mode automatically disables the *Upgrade* promotion button.**

> To turn this button off in Community mode, set the `promotion` option, which defaults to `true`, to false.

Reviewer 4: Doesn't having `which defaults to true` between `set` and `false` making this more confusing? Legitimate question.

Author: This has been re-written and the sub-clause is now a sentence in its own right.

Simple declarative sentences are, generally speaking, less likely to cause confusion. And hammering home the point through repetition is a hallmark of effective technical communications.

As Jazz bassist Adam Neely puts it, _repetition legitimises_.

[source,js]
----
tinymce.init({
  selector: "textarea",  // change this value according to your HTML
  promotion: false
});
----

> The *Upgrade* button appears as part of the {productname} menu bar and, consequently, does not appear if the menu bar is disabled.

Reviewer 4: Above, we said: 

> appears in the top-right of the editor menu bar

Should we keep the same phrasing?

Author: Works for me: done.

Reviewer 2: Will it matter that it's in not in the top-right for RTL modes? In that case it's in the top-left.

Author: > Will it matter that it's in not in the top-right for RTL modes? In that case it's in the top-left.

Good point.

How does _appears in the far corner of the editor menu bar_ read to you?

Author: I’ve gone, at least in this moment, for _unused corner_.

Reviewer 2: I have no strong opinions here, so whichever you think is best 😄

> IMPORTANT: It is not necessary to set `promotion` to `false` when {productname} is running in Premium mode. The *Upgrade* promotion button is disabled in this instance.

Reviewer 4: Would it be clearer with the injection of "automatically" 

> The *Upgrade* promotion button is automatically disabled in this instance

Also, integrator could still turn it on if they wanted to. Do we need the wording to convey that? Or just assume that it's so unlikely that trying to cover that case would just confuse the issue?

Author: _Automatically_ was injected as part of the re-write.

Author: > integrator could still turn it on if they wanted to.

After more thought and bad re-writes than I like to admit, I’ve added a new section, *Premium upgrade promotion option*.

It includes *Type*, *Default value* and *Possible values* entries and a NOTE explaining that it is possible to set this option contrary to its defaults no matter the distribution being used.

= Premium upgrade promotion
:navtitle: {productname} Premium upgrade promotion
:description: Editor options related to turning the Premium promotion display off
:keywords: upgrade, promotion, premium, button

== Premum upgrade promotion option

{productname} 6.2 and later includes the `promotion` option. It controls the presentation or otherwise of a Tiny-specific promotion button.

> 
> *Type:* `+Boolean+`
> 
> *Default value:* `+true+` in Community self-hosted instances; otherwise `+false+`. See xref:#premium-upgrade-promotion-defaults[immediately below] for details
> 
> *Possible values:* `+true+`, `+false+`
> 

Reviewer: ```suggestion
[[promotion]]

== `+promotion+`

*Type:* `+Boolean+`

*Default value:* `+true+` in Community self-hosted instances; otherwise `+false+`. See xref:#premium-upgrade-promotion-defaults[immediately below] for details

*Possible values:* `+true+`, `+false+`

== Example: Using `+promotion+`

[source,js]
----
tinymce.init({
  selector: "textarea",  // change this value according to your HTML
  promotion: false
});
----
```

Might be a good idea to turn this into a partial and include it here, to make it easier to include it elsewhere. I believe the option should be in `content-appearance.adoc` or `customize-ui.adoc` since they may be the first places users look.

Reviewer 2: @MitchCTiny I noticed this elsewhere in the other doc PRs, but this isn't how you do headings with ids. There should be no blank line between the id and heading. So it should be this instead:
```
[[promotion]]
== `+promotion+`
```

Author: Corrected as part of the ‘turn that section into a partial’ push just now.

Author: Also, turned that section into a partial and added `include::`s to the suggested pages.

Reviewer 2: Ahh good, I was going to raise it probably should have been a partial but thought I was being too picky.

The *Upgrade* promotion appears in the unused corner of the {productname} menu bar and, consequently, does not appear if the menu bar is disabled.

NOTE: these distribution-specific defaults are not fixed. Someone running {productname} as part of a Premium plan can turn this option on. And someone running the Community distribution of {productname} as a self-hosted instance can turn this option off.

[premium-upgrade-promotion-defaults]
== Premium upgrade promotion defaults

> When the Community distribution of {productname} 6.2 is running as a self-hosted instance, an *Upgrade* promotion button appears in the unused corner of the editor menu bar by default. The button does not appear when Community distributions of {productname} are running on the {cloudname}.
> 
> This promotion button redirects to a link:{companyurl}/tinymce-self-hosted-premium-features/[promotion page], showcasing the benefits of the variety of xref:plugins#premium-plugins[Premium Plugins] available for {productname}.
> 
> When {productname} 6.2 or later is running as part of a Premium plan, the *Upgrade* promotion button is disabled. And it is disabled when running with a Premium plan whether {productname} is running in the {cloudname} or as a self-hosted instance.
> 
> The {productname} Community distribution displays the promotion button when running as a self-hosted instance because the `promotion` option is set to `true` by default. To turn this button off in the Community disribution, set the `promotion` option to `false`.

Reviewer: ```suggestion
When the Community distribution of {productname} 6.2 is running as a self-hosted instance, an *Upgrade* promotion button appears in the unused corner of the editor menu bar by default. The button does not appear when Community distributions of {productname} are running on the {cloudname}. To turn this button off in the Community disribution, set the `promotion` option to `false`.

This promotion button redirects to a link:{companyurl}/tinymce-self-hosted-premium-features/[promotion page], showcasing the benefits of the variety of xref:plugins#premium-plugins[Premium Plugins] available for {productname}.

When {productname} 6.2 or later is running as part of a Premium plan, the *Upgrade* promotion button is disabled in both {cloudname} and self-hosted editors.
```

Author: The original is structured as follows:

1. The community distribution has the button.

2. What the button does.

3. Premium plans does not have the button.

4. Why community distribution has the button.

5. Removing the button from the community distribution.

6. NB re the button and Premium plans.

Your suggestion re-structures it thus:

1. The community distribution has the button.

2. Removing the button from the community distribution.

3. What the button does.

4. Premium plans does not have the button.

5. Why community distribution has the button.

6. NB re the button and Premium plans.

FWIW, I understand the impulse to bring *has the button* and *how to remove the button* together.

Nonetheless, I believe the original progressive reveal narrative structure flows better.

Bouncing from *remove the button* to *what the button does* feels like a small narrative jerk, to me.

[source,js]
----
tinymce.init({
  selector: "textarea",  // change this value according to your HTML
  promotion: false
});
----

IMPORTANT: It is not necessary to set `promotion` to `false` when {productname} is running as part of a Premium plan. Running {productname} as part of a Premium plan automatically disables the *Upgrade* promotion button.

> 
> 
> 
> 
> 
> 
> 

Reviewer: ```suggestion
```
Unnecessary lines

= Premium upgrade promotion
:navtitle: {productname} Premium upgrade promotion
:description: Editor options related to turning the Premium promotion display off
:keywords: upgrade, promotion, premium, button

> == Premum upgrade promotion option

Reviewer 2: ```suggestion
== Premium upgrade promotion option
```
Typo

Author: Corrected as part of the ‘turn that section into a partial’ push just now.

> {productname} 6.2 and later includes the `promotion` option. It controls the presentation or otherwise of a Tiny-specific promotion button.
> [[promotion]]
> 
> == `+promotion+`

Reviewer 2: ```suggestion
{productname} 6.2 and later includes the `promotion` option. It controls the presentation or otherwise of a Tiny-specific promotion button.

[[promotion]]
== `+promotion+`
```

Author: Corrected as part of the ‘turn that section into a partial’ push just now.
